cardinal/server/server_test.go

```diff
@@ -99,6 +99,84 @@ func makeTestTransactionHandler(t *testing.T, world *ecs.World, opts ...Option)
 	}
 }
 
+func TestShutDownViaMethod(t *testing.T) {
+	go func() {
+		select {
+		case <-time.After(10 * time.Second):
```

Comment:
The select portion of this is redundant.

```
go func() {
    <-time.After(10*time.Second)
    panic("took to long to shutdown"
} 
```
has the same behavior.

Reply:
yeah makes sense.

Reply:
I did some testing and I don't think this works as expected. Nothing cleans up this goroutine, so it continues to track time even after this test has exited. If this test is being run in isolation, it's fine, but if this test is being run with other tests (other tests in the package, or with the `--count` flag), it will trigger after the 10 seconds have elapsed.

I think this helper would do the trick:
```
func setTestTimeout(t *testing.T, timeout time.Duration) {
	if _, ok := t.Deadline(); ok {
		// A deadline has already been set. Don't add an additional deadline.
		return
	}
	success := make(chan bool)
	t.Cleanup(func() {
		success <- true
	})
	go func() {
		select {
		case <-success:
			// test was successful. Do nothing
		case <-time.After(timeout):
			assert.Check(t, false, "test timed out")
		}
	}()
}
```

Reply:
Actually the above code didn't terminate after timeout.

Reply:
You have to panic if it's in another green thread.

Reply:
Shoot. You could do a pattern like this: https://stackoverflow.com/questions/24929790/how-to-set-the-go-timeout-flag-on-go-test -- but you'll have to adjust your code in the 4 places that you're waiting for the server to start or stop.

Reply:
Or yeah, panic instead of assert.Check.